WooCommerce/Classes/POS/Utils/POSProductProvider.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+import class WooFoundation.CurrencySettings
+
+/// Product provider for the Point of Sale feature
+///
+final class POSProductProvider {
+    private let storageManager: StorageManagerType
+    private var siteID: Int64
+    private var currencySettings: CurrencySettings
+
+    init() {
+        self.storageManager = ServiceLocator.storageManager
+        self.siteID = ServiceLocator.stores.sessionManager.defaultSite?.siteID ?? 0
+        self.currencySettings = ServiceLocator.currencySettings
+    }
+
+    private lazy var productsResultsController: ResultsController<StorageProduct> = {
+        let predicate = NSPredicate(format: "siteID == %lld", siteID)
+        let descriptor = NSSortDescriptor(key: "name", ascending: true, selector: #selector(NSString.caseInsensitiveCompare))
+        let resultsController = ResultsController<StorageProduct>(storageManager: storageManager,
+                                                                  matching: predicate,
+                                                                  sortedBy: [descriptor])
+        return resultsController
+    }()
+
+    /// Provides a`[POSProduct]`array by mapping  simple, purchasable-only Products from storage
+    ///
+    func providePointOfSaleProducts() -> [POSProduct] {
+        var loadedProducts: [Product] = []
+        var pointOfSaleProducts: [POSProduct] = []
+
+        // 1. Fetch products from storage, and filter them by `purchasable` and `simple`
+        do {
+            try productsResultsController.performFetch()
+            if productsResultsController.fetchedObjects.isEmpty {
+                // TODO: Handle case for empty product list, or not empty but no eligible products
+                // https://github.com/woocommerce/woocommerce-ios/issues/12815
+                // https://github.com/woocommerce/woocommerce-ios/issues/12816
+                DDLogWarn("No products eligible for POS, or empty storage.")
+            } else {
+                // Ideally we should handle the filtering through a policy that can be easily modified,
+                // rather than having this declared implicitely:
```

Comment:
super nit:

```suggestion
                // rather than having this declared implicitly:
```

Reply:
Thanks! I fall in this typo way too often 😅